component/custom/custom_loader.go

```diff
@@ -0,0 +1,40 @@
+package custom
+
+import (
+	"os"
+	"plugin"
+	"strings"
+)
+
+func init() {
+	pluginPath := os.Getenv("PLUGIN_PATH")
+	if pluginPath == "" {
+		return
+	}
+
+	// environment variable can contain multiple paths separated by colon.
+	paths := strings.Split(pluginPath, ":")
+
+	for _, file := range paths {
+		println("custom_loader::Opening plugin:", file)
+		p, err := plugin.Open(file)
+		if err != nil {
+			println("custom_loader::Error opening plugin:", err.Error())
+			return
+		}
+		println("custom_loader::Plugin opened successfully")
+
+		initFunc, err := p.Lookup("InitPlugin")
```

Comment:
Better name ?

Reply:
Thinking more can we mandate the GetComponent method in every external component. Then the custom_loader can call GetComponent to get an instance of the component, verify that it is of type external.Component and then call internal.AddComponent with the GetComponent method as init

Reply:
That way nobody external can call AddComponent and you can remove it from the external.go

Reply:
yup makes more sense.